userprofile/templates/userprofile/login.html

```diff
@@ -20,84 +20,130 @@ <h1>{% trans "Login" %}</h1>
-							<h3>
+							<h2>
 								{{ shibboleth_title_text|safe }}
-							</h3>
+							</h2>
 						{% endif %}
 						{% if shibboleth_body_text %}
 							<p>
 								{{ shibboleth_body_text|safe }}
 							</p>
 						{% endif %}
 						<p>
-							<a class="aplus-button--secondary aplus-button--lg" role="button" href="{% url 'shibboleth-login' %}?next={{ next|iriencode }}">
+							<a class="aplus-button--secondary aplus-button--lg" role="button"
+								href="{% url 'shibboleth-login' %}?next={{ next|iriencode }}">
 								{{ shibboleth_button_text|safe }}
 							</a>
 						</p>
 					</div>
 				</div>
-			</div>
+			</section>
 		{% endif %}
 
 		{% if mooc_login %}
-			<div class="col-sm-6 col-md-3 login-box">
-				<div class="thumbnail">
-					<div class="caption">
+			<section aria-label="{{ mooc_title_text|safe }} login" class="col-sm-6 col-md-3 login-box">
+				<div class="login-section">
+					<div class="login-section-body">
 						{% if mooc_title_text %}
-							<h3>
+							<h2>
 								{{ mooc_title_text|safe }}
-							</h3>
+							</h2>
 						{% endif %}
 						{% if mooc_body_text %}
 							<p>
 								{{ mooc_body_text|safe }}
 							</p>
 						{% endif %}
 						<p>
-							<a class="aplus-button--secondary aplus-button--lg" role="button" href="{% url 'social:begin' 'google-oauth2' %}?next={{ next|iriencode }}">{% trans 'Log in using Google' %}</a>
+							<a class="aplus-button--secondary aplus-button--lg" role="button"
+								href="{% url 'social:begin' 'google-oauth2' %}?next={{ next|iriencode }}">{% trans 'Log in using Google' %}</a>
 						</p>
 					</div>
 				</div>
-			</div>
+			</section>
 		{% endif %}
 
-		<div class="col-sm-6 col-md-3 login-box extra-login">
-			<div class="thumbnail">
-				<div class="caption">
-					<h3>
-						{% trans "Local users" %}
-					</h3>
-					<p>
-						{% blocktrans trimmed %}
-							If you have been provided with credentials specifically
-							for {{ brand_name }}, use this login.
-						{% endblocktrans %}
-					</p>
-					<form method="post" role="form">
-						{% csrf_token %}
-						{{ form|bootstrap }}
+	<section aria-label="{% trans 'Local users' %} login" class="col-sm-6 col-md-3 login-box extra-login">
+		<div class="login-section">
+			<div class="login-section-body">
+				<h2>
+					{% trans "Local users" %}
+				</h2>
+				<p>
+					{% blocktrans trimmed %}
+						If you have been provided with credentials specifically
+						for {{ brand_name }}, use this login.
+					{% endblocktrans %}
+				</p>
+				<form method="post" role="form">
+					{% csrf_token %}
+						{% if form.non_field_errors %}
+							<div class="alert alert-danger">
+								<a class="close" aria-label="Close error list" data-dismiss="alert">×</a>
+									{% for error in form.non_field_errors %}
+										{{ error|escape }}
+									{% endfor %}
+							</div>
+						{% endif %}
+						{% if form.username.errors %}
+							{% for error in form.username.errors %}
```

Comment:
Doesn't the error loop render the username input multiple times for each error? Usually, the error loop only renders the errors and not the input. The input could be rendered after the loop.